src/main/java/com/petfinder/controller/AdvertisementController.java

```diff
@@ -54,55 +69,201 @@ public String newAdvertisement(@RequestParam(required = false) String title,
+                        new FileOutputStream(new File(newFileName)));
+                FileCopyUtils.copy(file.getInputStream(), stream);
+                stream.close();
+                LOGGER.log(Level.SEVERE, "You successfully uploaded {0}!", file.getOriginalFilename());
+                return newFileName;
+            } catch (Exception e) {
+                LOGGER.log(Level.SEVERE, "You failed to upload {0} => {1}", new Object[]{file.getOriginalFilename(), e.getMessage()});
+            }
+        } else {
+            LOGGER.log(Level.SEVERE, "You failed to upload {0} because the file was empty", file.getOriginalFilename());
+        }
+        return null;
+    }
+
+    @RequestMapping(value = "/getNewAdd")
+    public String getToNewAdvertisement(Model model) {
+        model.addAttribute("categories", advertisementService.getAllCategories());
         return "addAdvertisement";
     }
-	
-	@ResponseBody
-	@RequestMapping(value = "/searchResult")
-	public ModelAndView getSearchResults( Model model,
-        @RequestParam(required = false) String adInfo,
-        @RequestParam(required = false) String petInfo,
-        @RequestParam(required = false) String locationInfo,
-        @RequestParam(required = false) String tagInfo,
-        @RequestParam(required = false) int page
-	) {
+
+    private List<Tag> tagStringToList(String tagsString) {
+        List<String> listTags = new ArrayList<>(Arrays.asList(tagsString.split(",")));
+        listTags.removeAll(Arrays.asList("", null));
+        List<Tag> tags = new ArrayList<>();
+        for (String listTag : listTags) {
+            tags.add(new Tag(listTag.trim()));
+        }
+        return tags;
+    }
+
+    @RequestMapping(value = "/editAdd")
+    public String editAdvertisement(@RequestParam(required = false) Long id, Model model,
+            @RequestParam(required = false) String title,
+            @RequestParam(required = false) String content,
+            @RequestParam(required = false) String petName,
+            @RequestParam(required = false) Integer age,
+            @RequestParam(required = false) String race,
+            @RequestParam(required = false) String categoryName,
+            @RequestParam(required = false) String voivodership,
+            @RequestParam(required = false) String commune,
+            @RequestParam(required = false) String place,
+            @RequestParam(required = false) String tagsString,
+            @RequestParam(required = false) MultipartFile image,
+            @RequestParam(required = false) MultipartFile video) {
+
+        if (title != null && content != null && petName != null && race != null && categoryName != null && voivodership != null && commune != null && place != null) {
+            if (!title.equals("") && !content.equals("") && !petName.equals("") && !race.equals("") && !categoryName.equals("") && !voivodership.equals("") && !commune.equals("") && !place.equals("")) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/matikz1110/PetFinder/file/2653733612/issues/source?bid=3239464&fileBranchId=3326233#l163)